Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM nvidia/cuda:10.1-devel-ubuntu18.04
+
+WORKDIR /
+
+# Package and dependency setup
+RUN apt-get update \
+    && apt-get -y install software-properties-common \
+    && apt-get update \
+    && apt-get install -y git cmake build-essential
+
+# Add source files
+ADD . /ethminer
```

Comment:
This might be fine from the developer perspective but probably the right thing is to clone from git a specific tag or branch.
Also copying everything from the current directory might cause some issues when building (e.g. garbage from local builds). Usually the recommendation is to copy explicitly what it's needed rather than everything but I also understand this might not be suitable either.